src/datepicker/datepicker.ts

```diff
@@ -388,7 +388,9 @@ export class NgbDatepicker implements OnDestroy,
           .pipe(
               filter(
                   ({target, relatedTarget}) =>
-                      !(hasClassName(target, 'ngb-dp-day') && hasClassName(relatedTarget, 'ngb-dp-day'))),
+                      !(hasClassName(target, 'ngb-dp-day') && hasClassName(relatedTarget, 'ngb-dp-day') &&
+                        this._elementRef.nativeElement.contains(target as Node) &&
+                        this._elementRef.nativeElement.contains(relatedTarget as Node))),
```

Comment:
Please create a local reference `const {nativeElement} = this._elementRef;` outside of the Observable chain.